src/types/types.ts

```diff
@@ -1,12 +1,35 @@
+export enum Region {
+  Africa = 'Africa',
+  Americas = 'Americas',
+  Asia = 'Asia',
+  Europe = 'Europe',
+  Oceania = 'Oceania',
+}
+
 export interface Country {
   name: {
     common: string;
   };
   population: number;
-  region: string;
+  region: Region;
   capital?: string;
   cca3: string;
   flags: {
     svg: string;
   };
 }
+
+export interface CountryDetails extends Country {
+  subregion: string;
+  borders?: string[];
+  currencies: {
+    [key: string]: {
```

Comment:
a tu byłoby Record<string, { name: string; symbol: string;}>, ale to tylko inna możliwość, mapped types też są w porządku :D